### DISCUSSION TITLE
How to create a custom Pixel Format?

### DISCUSSION THREAD

**Discussion by Jacbo1** [SixLabors/ImageSharp#2274] [Q&A] [answered]
How do I implement a custom pixel format for R8? The docs say the pixel formats are extensible so I looked at the source code and tried implement an R8 pixel format by copying and adapting the A8 struct but then found out 2 of the methods called in the A8 struct are internal i.e. cannot be accessed from an implementing project. At least one of these is required for it to work. `new PixelOperations()` required as part of implementing `IPixel<R8>` and `PixelTypeInfo.Create<R8>()` (or maybe `PixelTypeInfo` itself) and internal only.

**Comment by antonfirsov** [comment, marked as answer]
> `new PixelOperations()` required

The `PixelOperations<T>` type is not abstract, and the constructor is public, you should be able to instantiate a `new PixelOperations<R8>()`. Subclassing is needed to optimize conversion operations: you can create your own `R8PixelOperations : PixelOperations<R8>` subclass and override methods with optimized ones.

> `PixelTypeInfo.Create<R8>()` (or maybe `PixelTypeInfo` itself)

You can use one of the [public constructors](https://github.com/SixLabors/ImageSharp/blob/7b403897dfee6433d7c39feb6fdef1f4dfe0949b/src/ImageSharp/Formats/PixelTypeInfo.cs#L18-L28) to instantiate a `PixelTypeInfo`. `Create<TPixel>` is really just a [thin wrapper](https://github.com/SixLabors/ImageSharp/blob/7b403897dfee6433d7c39feb6fdef1f4dfe0949b/src/ImageSharp/Formats/PixelTypeInfo.cs#L45-L47) around them.
